internal/scheduler/scheduler.go

```diff
@@ -0,0 +1,181 @@
+// Package scheduler implements a simple scheduler that triggers the next item when its due time is
+// reached based on the list of upcoming items.
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/internal/ctxutil"
+	"github.com/kopia/kopia/repo/logging"
+)
+
+var log = logging.Module("scheduler")
+
+// GetItemsFunc is a callback that returns items for the scheduler to consider.
+type GetItemsFunc func(ctx context.Context, now time.Time) []Item
+
+// Item is a describes item that can be scheduled with a function invoked the next time the item is due.
+type Item struct {
+	Description string
+	NextTime    time.Time
+	Trigger     func()
+}
+
+// Scheduler is a scheduler triggers the next upcoming events based on the list of items returned by listItems().
```

Comment:
this comment is confusing, please reword and clarify.

Reply:
fixed in previous PR